### PR TITLE
Add Authenticator#returnAttributes.

### LINKED ISSUE #89: Allow configuration of return attributes on the Authenticator
Return attributes are defined as part of the authentication request.
It would be useful if those attributes could be supplemented with additional attributes that applies to all users.
In particular, several password policy implementation require that certain attributes be returned.

Update ```Authenticator``` with a returnAttributes property.

### PR DESCRIPTION
**Pull request by dfish3r**
Add returnAttributes for each authentication response handler.
Update spring parser to set appropriate return attributes per authentication response handler.
Fixes #89.

### CODE REVIEW

File: core/src/main/java/org/ldaptive/auth/ext/ActiveDirectoryAuthenticationResponseHandler.java [hunk 1]
```diff
@@ -115,9 +117,6 @@ public Period getExpirationPeriod()
    */
   public void setExpirationPeriod(final Period period)
   {
-    if (period == null) {
-      throw new IllegalArgumentException("Expiration cannot be null");
-    }
```
Comment (serac) [inline core/src/main/java/org/ldaptive/auth/ext/ActiveDirectoryAuthenticationResponseHandler.java]: Why did you remove this null check? Seems unrelated to the refactoring described in the issue.

File: core/src/main/java/org/ldaptive/auth/ext/ActiveDirectoryAuthenticationResponseHandler.java [hunk 2]
```diff
@@ -140,9 +139,6 @@ public Period getWarningPeriod()
    */
   public void setWarningPeriod(final Period period)
   {
-    if (period == null) {
-      throw new IllegalArgumentException("Warning cannot be null");
-    }
```
Comment (serac) [inline core/src/main/java/org/ldaptive/auth/ext/ActiveDirectoryAuthenticationResponseHandler.java]: Same as above.

File: core/src/main/java/org/ldaptive/auth/ext/EDirectoryAuthenticationResponseHandler.java [hunk 3]
```diff
@@ -89,9 +92,6 @@ public Period getWarningPeriod()
    */
   public void setWarningPeriod(final Period period)
   {
-    if (period == null) {
-      throw new IllegalArgumentException("Warning cannot be null");
-    }
```
Comment (serac) [inline core/src/main/java/org/ldaptive/auth/ext/EDirectoryAuthenticationResponseHandler.java]: Same as above.

File: core/src/main/java/org/ldaptive/auth/ext/FreeIPAAuthenticationResponseHandler.java [hunk 4]
```diff
@@ -158,9 +165,6 @@ public Period getExpirationPeriod()
    */
   public void setExpirationPeriod(final Period period)
   {
-    if (period == null) {
-      throw new IllegalArgumentException("Expiration cannot be null");
-    }
```
Comment (serac) [inline core/src/main/java/org/ldaptive/auth/ext/FreeIPAAuthenticationResponseHandler.java]: Same as above

File: core/src/main/java/org/ldaptive/auth/ext/FreeIPAAuthenticationResponseHandler.java [hunk 5]
```diff
@@ -183,9 +187,6 @@ public Period getWarningPeriod()
    */
   public void setWarningPeriod(final Period period)
   {
-    if (period == null) {
-      throw new IllegalArgumentException("Warning cannot be null");
-    }
```
Comment (serac) [inline core/src/main/java/org/ldaptive/auth/ext/FreeIPAAuthenticationResponseHandler.java]: Same as above.

File: core/src/main/java/org/ldaptive/auth/ext/ActiveDirectoryAuthenticationResponseHandler.java (hunk 1 above)
Comment (dfish3r) [inline core/src/main/java/org/ldaptive/auth/ext/ActiveDirectoryAuthenticationResponseHandler.java]: The check is an artifact of the conversion to ```Period```, when negatives values were not allowed.
It's no longer correct, these variables can be null.

I foresee a use case where configuration is using the full constructor and allowing nulls to pass through will make for easier wiring.

**Comment by serac** [comment]
Looks good generally. I noted the removal of several null checks, which appears to be an unrelated change but perhaps is justifiable.

**Comment by dfish3r** [comment]
Ok. Merge if you are satisfied.
